### ISSUE TITLE
Wonder Trade bot unreliable behaviour

### ISSUE THREAD

**Issue by ghost** [drgoku282/PKMN-NTR#154]
Tested on old 3DS, all installed firmware and software is up-to-date to their latest revisions.

Running bot with following settings:

![wonder-trade-bot](https://user-images.githubusercontent.com/15217150/37192630-e46ecbae-2366-11e8-82ad-5fe8b5f97ac2.png)

With only one Pokémon present in the folder.

The following behaviour can occur after a few trades:

- The bot will leave the Wonder Trade Start screen instead of proceeding to the box.
- The Pokémon in the current selection slot gets traded instead of the one in the Wonder Trade folder.
- The bot will error asking to return to the Wonder Trade Start screen and to start again.
- The device will freeze permanently on a "saving" screen. 

Tested on Pokémon Ultra Moon v1.2.

**Comment by ghost** [comment]
It seems the 30 second bot wait once a trade is starting is too little. I think raising it to 32 might fix this issue, or 35 if you really want to be safe.

**Comment by ghost** [comment]
I compiled it with a 40 second time out, it seems to be safe. The ultimate solution would be to just detect when it gets back to the start screen of course, without having to deal with risky time out values.

35 seconds even caused issues, because sometimes when saving after a trade, it can happen quickly, or it can take a few seconds extra, and when the bot tries to continue then, it will result in the game to freeze entirely.

TL;DR: 40 seconds timeout instead of 30 seconds (or remove timeout and detect when back at start)

**Comment by D35P4C1T0** [comment]
Do you mind sharing your 40 seconds build?
